ibm/resource_ibm_private_dns_glb_pool.go

```diff
@@ -0,0 +1,340 @@
+	pdnsGlbPoolCreatedOn               = "created_on"
+	pdnsGlbPoolModifiedOn              = "modified_on"
+)
+
+func resourceIBMPrivateDNSGLBPool() *schema.Resource {
+	return &schema.Resource{
+
+		Create:   resourceIBMPrivateDNSGLBPoolCreate,
+		Read:     resourceIBMPrivateDNSGLBPoolRead,
+		Update:   resourceIBMPrivateDNSGLBPoolUpdate,
+		Delete:   resourceIBMPrivateDNSGLBPoolDelete,
+		Exists:   resourceIBMPrivateDNSGLBPoolExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			pdnsInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Instance Id",
+			},
+			pdnsGlbPoolID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Monitor Id",
+			},
+			pdnsGlbPoolName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique identifier of a service instance.",
+			},
+			pdnsGlbPoolDescription: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Descriptive text of the load balancer pool",
+			},
+			pdnsGlbPoolEnabled: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Whether the load balancer pool is enabled",
+			},
+			pdnsGlbPoolHealthyOriginsThreshold: {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The minimum number of origins that must be healthy for this pool to serve traffic",
+			},
+			pdnsGlbPoolOrigins: {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "Origins info",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						pdnsGlbPoolOriginsName: {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						pdnsGlbPoolOriginsAddress: {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						pdnsGlbPoolOriginsEnabled: {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						pdnsGlbPoolOriginsDescription: {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			pdnsGlbPoolMonitor: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The ID of the load balancer monitor to be associated to this pool",
+			},
+			pdnsGlbPoolChannel: {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"webhook"}),
+				Description:  "The notification channel",
+			},
+			pdnsGlbPoolRegion: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"us-south", "us-east", "eu-gb", "eu-du", "au-syd", "jp-tok"}),
+				Description:  "Health check region of VSIs",
+			},
+			pdnsGlbPoolSubnet: {
```

Comment:
this is a string slice.  please change it and add attribute in the test case for testing it.

Reply:
handled